src/human-readable/runtime/utils.ts

```diff
@@ -129,11 +129,13 @@ type ParseOptions = {
   modifiers?: Modifier | readonly Modifier[]
   structs?: StructLookup
   type?: 'constructor' | 'error' | 'event' | 'function'
+  parseContext?: 'structs'
 }
 
 export function parseAbiParameter(param: string, options?: ParseOptions) {
   // optional namespace cache by `type`
   const paramKey = `${options?.type ?? ''}${param}`
```

Comment:
Didn't realize how funky this was before. Might be a good idea to turn this into a function inside of `cache.ts` (e.g. `getParameterCacheKey`)

```suggestion
  const paramKey = `${options?.type ? `${options?.type}:` : ''}${param}`
```

Reply:
Yeah i agree. Makes things cleaner.